h2o-core/src/main/java/water/jdbc/SQLManager.java

```diff
@@ -171,39 +185,62 @@ SELECT ROW_NUMBER() OVER () AS RowNum_, <table>.* FROM <table>
     double binary_ones_fraction = 0.5; //estimate
 
     //create template vectors in advance and run MR
-    long totSize =
+    final long totSize =
             (long)((float)(catcols+intcols)*numRow*4 //4 bytes for categoricals and integers
                     +(float)bincols          *numRow*1*binary_ones_fraction //sparse uses a fraction of one byte (or even less)
                     +(float)(realcols+timecols+stringcols) *numRow*8); //8 bytes for real and time (long) values
-    final Vec _v;
+
+    final Vec _retrieval_v;
+    final int chunk_size = FileVec.calcOptimalChunkSize(totSize, numCol, numCol * 4,
+            H2O.ARGS.nthreads, H2O.getCloudSize(), false, false);
+    final double rows_per_chunk = chunk_size; //why not numRow * chunk_size / totSize; it's supposed to be rows per chunk, not the byte size
+    final Vec _v = makeCon(0, numRow, (int) Math.ceil(Math.log1p(rows_per_chunk)), false);
```

Comment:
I would remove the leading underscore from `_v` and `_retrieval_v`. It goes against the coding conventions and this is a good opportunity to fix it.

Reply:
done

Reply:
removing it for other attributes in this file as well